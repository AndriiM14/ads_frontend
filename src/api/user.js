import axios from 'axios';
import { DEFAULT_HEADERS, formAuthHeaders } from './config';

export const createUser = (data) => axios.post('/registration', data, { headers: DEFAULT_HEADERS });

export const getBootstrap = () => axios.get('/user', { headers: formAuthHeaders() });

export const getUsers = () => axios.get('/users', { headers: formAuthHeaders() });

export const getUser = (id) => axios.get(`/users/${id}`, { headers: formAuthHeaders() });

export const deleteUser = (id) => axios.delete(`/users/${id}`, { headers: formAuthHeaders() });

export const changeUser = (id, data) => axios.put(`/users/${id}`, data, { headers: formAuthHeaders() });
