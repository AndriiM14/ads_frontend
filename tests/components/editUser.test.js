import React from "react";
import { mockUser } from '../mock/user';
import { render, screen, fireEvent  } from "@testing-library/react";
import EditUser from "../../src/components/editUser";

describe('EditUser component tests', () => {
    test("Proper data render", () => {
        const data = mockUser;

        const { asFragment } = render(
            <EditUser data={data} onConfirm={jest.fn()} onCancel={jest.fn()} />
        );

        const emailInput = screen.getByTestId("email");
        const confirmButton = screen.getByTestId("confirm");
        const cancelButton = screen.getByTestId("cancel");

        const newMail = "newmail@mail.com";
        fireEvent.change(emailInput, { target: { value: newMail } });

        expect(emailInput.value).toBe(newMail);

        fireEvent.click(confirmButton);
        fireEvent.click(cancelButton);

        expect(asFragment).toMatchSnapshot();
    });
});