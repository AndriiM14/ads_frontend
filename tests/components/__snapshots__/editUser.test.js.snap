// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUser component tests Proper data render 1`] = `[Function]`;
